synapse/storage/registration.py

```diff
@@ -86,6 +87,26 @@ def get_user_by_access_token(self, token):
             token
         )
 
+    @defer.inlineCallbacks
```

Comment:
I'd stick a `@cachedInlineCallbacks()` here so that we cache the result, we'll also need to invalidate it when we update it.